reactor-pool/src/main/java/reactor/pool/AbstractPool.java

```diff
@@ -441,14 +443,22 @@ public void request(long n) {
 		 * Stop the countdown started when calling {@link AbstractPool#doAcquire(Borrower)}.
 		 */
 		void stopPendingCountdown(boolean success) {
-			if (!timeoutTask.isDisposed()) {
-				if (success) {
-					pool.metricsRecorder.recordPendingSuccessAndLatency(pool.clock.millis() - pendingAcquireStart);
-				} else {
-					pool.metricsRecorder.recordPendingFailureAndLatency(pool.clock.millis() - pendingAcquireStart);
-				}
+			if (timeoutTask == null) {
+				return;
 			}
+
 			timeoutTask.dispose();
+			timeoutTask = null;
+
+			if (pendingAcquireStart == 0) {
+				return;
+			}
+
+			if (success) {
+				pool.metricsRecorder.recordPendingSuccessAndLatency(pool.clock.millis() - pendingAcquireStart);
+			} else {
+				pool.metricsRecorder.recordPendingFailureAndLatency(pool.clock.millis() - pendingAcquireStart);
+			}
```

Comment:
@HeartPattern , first, thanks for your contribution;

I think it's ok, but I wonder if we can reduce the size of the changes done in the `stopPendingCountdown` method.
What do you think about doing the following in the stopPendingCountdown method ?

```suggestion
			if (pendingAcquireStart > 0) {
				if (success) {
					pool.metricsRecorder.recordPendingSuccessAndLatency(pool.clock.millis() - pendingAcquireStart);
				} else {
					pool.metricsRecorder.recordPendingFailureAndLatency(pool.clock.millis() - pendingAcquireStart);
				}
				pendingAcquireStart = 0;
			}
			timeoutTask.dispose();
```

Reply:
That seems great. Added new commit.